test/l1l2/l1_tx.ts

```diff
@@ -0,0 +1,228 @@
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { expect } from 'chai';
+import { Contract, ContractFactory } from 'ethers';
+import { starknet, network, ethers } from 'hardhat';
+import { FOR, SplitUint256 } from '../starknet/shared/types';
+import { StarknetContractFactory, StarknetContract, HttpNetworkConfig } from 'hardhat/types';
+import { strToShortStringArr } from '@snapshot-labs/sx';
+import { getCommit } from '../starknet/shared/helpers';
+
+const propose_selector = BigInt(
+  '0x1BFD596AE442867EF71CA523061610682AF8B00FC2738329422F4AD8D220B81'
+);
+const vote_selector = BigInt('0x132BDF85FC8AA10AC3C22F02317F8F53D4B4F52235ED1EABB3A4CBBE08B5C41');
+const VOTING_DELAY = BigInt(0);
+const VOTING_PERIOD = BigInt(20);
+const RANDOM_ADDRESS = BigInt('0xAD4Eb63b9a2F1A4D241c92e2bBa78eEFc56ab990');
+
+export async function setup(signer: SignerWithAddress) {
```

Comment:
We will be able to move this function to the `setup.ts` file once the other PR gets merged :D